### PR TITLE
apply vscode powershell formatting

### LINKED ISSUE #207: Code style?
## 1. Provide a general summary of the issue in the Title above

Looking through the project, I see the inconsistency in the code style, I would have a hard time contributing without agreeing on a code style.

No indents: https://github.com/ngetchell/PSGitLab/blob/master/PSGitLab/Private/DecryptString.ps1
2 Space indents with varied spacing too: https://github.com/ngetchell/PSGitLab/blob/master/PSGitLab/Private/DownloadFromGitLabAPI.ps1
4 space idents: https://github.com/ngetchell/PSGitLab/blob/master/PSGitLab/Private/FormatCommits.ps1

So what is? Would love to add an `.editorconfig` and some PR quality checks.

### PR DESCRIPTION
**Pull request by jetersen**
Goes to show the level of inconsistency :smile: 

fixes #207

### CODE REVIEW

**Comment by ngetchell** [comment]
Does the editorconfig VS Code extension provide something that the PowerShell extension doesn't when it comes to formatting? I think I prefer the Format Document function over the installation of another extension. It is language aware and visually appealing. 

I'm not going to accept a PR that requires me to test every file. I didn't invest enough time in automated tests to make sure these whitespace changes won't break anything. I simply don't have the time to do it by hand. 

The commit changing the if elseif elseif..... to a switch would be accepted as a PR.

File: PSGitLab/Private/FormatCommits.ps1 [hunk 1]
```diff
@@ -50,20 +50,14 @@ Function FormatCommits {
...
-        } elseif ($day.Name -eq 6) {
-          $output.Sat = $day.Count
+        switch ($day.Name) {
+          0 { $output.Sun = $day.Count }
+          1 { $output.Mon = $day.Count }
+          2 { $output.Tue = $day.Count }
+          3 { $output.Wed = $day.Count }
+          4 { $output.Thu = $day.Count }
+          5 { $output.Fri = $day.Count }
+          6 { $output.Sat = $day.Count }
```
Comment (ngetchell) [inline PSGitLab/Private/FormatCommits.ps1]: There should be a default case, even if it only exists to throw an error.

**Review by ngetchell — CHANGES_REQUESTED** [review]
Is another vs code extension (above the ps ext) needed to perform this or can it be done within the Format Document functionality inside vs code?

**Comment by jetersen** [comment]
@ngetchell `.editorconfig` is used to tell the user and their editor what is expected.

#198 added tabs for instance in a new file, which furthers the inconsistency.

Of course, if the user uses an editor without support for `.editorconfig` it is no good, but then we ask CI to check.

**Comment by jetersen** [comment]
In the process of adding PSScriptAnalyze with some extra rules to have Appveyor run those :)

Current count is 2225 (Reverted format commit due to merges :wink:)

**Comment by jetersen** [comment]
This should fail on appveyor due to linting and the next commit should hopefully fix most of it and then the next commit will fix the rest :smile:

**Comment by jetersen** [comment]
@ngetchell what do you think about `alignPropertyValuePairs` rule?  
The current standard looked like no alignment.  
Which meant I had to commit `.vscode/settings.json` for that since it is the default.

Plus whitespace consistency and `alignPropertyValuePairs` in PSScriptAnalyzer seems to have a conflict at the moment.

**Comment by it-praktyk** [comment]
I like that change!

**Comment by ngetchell** [comment]
I will update the files as I go with the VS Code formatter but I am not interested in a PR this large to make sweeping changes to the module.

**Pull request closed without merging**